VisualizationBase/src/nodes/InfoNode.cpp

```diff
@@ -73,11 +73,19 @@ void InfoNode::updateInfo(bool isAutoUpdate)
 			cachedInfoStrings_[name] = getter.getter_(target_);
 		//Use the computed value to extend the info
 		if (!cachedInfoStrings_[name].isEmpty())
-			infoHtml += cachedInfoStrings_[name] + "<small>" + "  (Layer " + name + ")</small>" + "<br>";
+			infoHtml += cachedInfoStrings_[name] + "<small>" + "  (Layer " + name + ")</small>" + "<br><hr>";
 	}
+	infoHtml = "<html>" + infoHtml + "</div></html>";
 	setInfoHtml(infoHtml);
 }
 
+void InfoNode::setInfoHtml(QString content)
+{
+	if (content == infoHtml_) return;
+	infoHtml_ = content;
+	contentChanged_ = true;
```

Comment:
Hmm, actually I realize now that this mechanism is a bit brittle. Let's try something else.
- Instead of introducing the `contentChanged_` flag, let's reuse some existing functionality: the node's revision.
- So remove `contentChanged`, and instead here, if you detect a difference, just call `incrementRevision()`. Every node has a revision, which is just a number that tracks the current version of the node, so we just indicate that we've changed the content.
- Changing the revision means that the node will automatically be updated.